src/App.tsx

```diff
@@ -1,54 +1,89 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+
+const debounce = (func:React.Dispatch<React.SetStateAction<string>>,
+  delay:number) => {
+  let timerId: number;
+
+  return (...args:string[]) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(func, delay, ...args);
+  };
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selected, setSelected] = useState<Person>();
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
+
+  const visiblePeople = useMemo(() => {
+    return peopleFromServer.filter(
+      person => person.name.toLowerCase().includes(query.toLowerCase()),
+    );
+  }, [peopleFromServer, appliedQuery]);
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selected
+          ? `${selected.name} (${selected.born} = ${selected.died})`
+          : 'No selected person'}
       </h1>
 
-      <div className="dropdown is-active">
+      <div
+        className={appliedQuery ? 'dropdown is-active' : 'dropdown'}
+      >
         <div className="dropdown-trigger">
           <input
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={query}
+            onChange={event => {
+              setQuery(event.target.value);
+              applyQuery(event.target.value);
+            }}
           />
         </div>
 
         <div className="dropdown-menu" role="menu">
           <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
+            {visiblePeople.length >= 1
```

Comment:
Pls [see the doc](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)
```suggestion
            {visiblePeople.length
```